src/adapter/src/catalog.rs

```diff
@@ -6679,6 +6693,22 @@ impl SessionCatalog for ConnCatalog<'_> {
     fn aws_privatelink_availability_zones(&self) -> Option<BTreeSet<String>> {
         self.state.aws_privatelink_availability_zones.clone()
     }
+
+    fn get_feature(&self, name: CatalogFeature) -> bool {
+        use CatalogFeature::*;
+        let config = &self.state.system_configuration;
+        match name {
+            EnableWithMutuallyRecursive => config.enable_with_mutually_recursive(),
+        }
+    }
+
+    fn set_feature(&mut self, name: CatalogFeature, value: bool) -> bool {
+        use CatalogFeature::*;
+        let config = &mut self.state.to_mut().system_configuration;
+        match name {
+            EnableWithMutuallyRecursive => config.set_enable_with_mutually_recursive(value),
```

Comment:
As per the other conversation, this modifies the catalog state, which is used by all current and future sessions.

Reply:
I thought that `self.state.to_mut()` is copy-on-write, so it will just clone the state for this `ConnCatalog` instance. Let me test this.

Reply:
> I thought that `self.state` is copy-on-write (`Cow<_>`), so `self.state.to_mut()` will just clone the state for this `ConnCatalog` instance. Let me test this.

Oh, you're absolutely right. I didn't realize that it's a `Cow`. This should be fine then.

Reply:
I tested and after I restart `bin/environmentd` a catalog entry that uses WMR is re-created, but 
```sql
show enable_with_mutually_recursive;
```
still returns `false`. If @antiguru wants this behavior I will add an `mzcompose` test for this scenario.

Reply:
> Also this is the only place where we modify a SystemVar in memory, without also modifying it in the stash. That's probably fine but we should call this out in some doc-comment with an explanation as to why this is.

I'll extend the `set_feature` docs so people are not tricked of assuming that these calls are being persisted.